### PR TITLE
Sanlouise 14741 cleanup veteranstatus

### PR DESCRIPTION
**Pull request by sanlouise**
## Description
At some point, `isVeteran` was duplicated thrice in the redux store and these duplications have been around for a very long time. Applications have become dependent on those duplicated properties, and those duplications and dependencies are cleaned up by this PR. I realize this is touching integral logic to our application and code that has not been touched in years. However, I could not at least give it a try to clean up this little mess. This update should be tested very thoroughly before merging.

To see the structure below, log into staging with user +36.

image

I believe we want the `veteranStatus` object (and the 'status' on the profile which indicated whether the user in in `MVI`) in our redux store to look like:

```
 veteranStatus {
   status: "OK",
   isVeteran: true,
   servedInMilitary: true,
 }
status: "OK",
```

and no longer like:

```
 isVeteran: true,
 veteranStatus {
   isVeteran: true,
   veteranStatus {
     status: "OK",
     isVeteran: true,
     servedInMilitary: true,
   },
   servedInMilitary: true,
 }
 status: "OK",
```

### Redux store AFTER (local, user +1)
image

## Testing done
All works well locally, updated tests where needed.

## Acceptance criteria
- [x] Clean up `veteranStatus` in the redux store 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by micahchiang — APPROVED** [review]

File: src/applications/personalization/rated-disabilities/tests/components/RatedDisabilityView.unit.spec.jsx [hunk 1]
```diff
@@ -8,7 +8,9 @@ describe('<RatedDisabilityView/>', () => {
   const user = {
     profile: {
       verified: true,
-      status: 'OK',
```
Comment (erikphansen) [inline src/applications/personalization/rated-disabilities/tests/components/RatedDisabilityView.unit.spec.jsx:11]: this one should stay as is; `profile.status` is still a thing and different fron `profile.veteranStatus.status`

File: src/applications/veteran-id-card/containers/VeteranIDCard.jsx [hunk 2]
```diff
@@ -40,9 +40,9 @@ class VeteranIDCard extends React.Component {
         if (serviceRateLimitedAuthed) {
           recordEvent({ event: 'vic-authenticated-ratelimited' });
           this.renderEmailCapture = true;
-          if (userProfile.veteranStatus === 'NOT_FOUND') {
+          if (userProfile.veteranStatus.status === 'NOT_FOUND') {
```
Comment (erikphansen) [inline src/applications/veteran-id-card/containers/VeteranIDCard.jsx:43]: Probably not needed based on how you are giving `veteranStatus` a default value of `{}`, but to be extra careful, should likely do `userProfile.veteranStatus?.status` here...

File: src/applications/veteran-id-card/containers/VeteranIDCard.jsx [hunk 3]
```diff
@@ -40,9 +40,9 @@ class VeteranIDCard extends React.Component {
...
             recordEvent({ events: 'vic-emis-lookup-failed' });
-          } else if (userProfile.veteranStatus === 'SERVER_ERROR') {
+          } else if (userProfile.veteranStatus.status === 'SERVER_ERROR') {
```
Comment (erikphansen) [inline src/applications/veteran-id-card/containers/VeteranIDCard.jsx:45]: ...and here as well.

File: src/platform/user/profile/utilities/index.js [hunk 4]
```diff
@@ -88,20 +88,16 @@ export function mapRawUserDataToState(json) {
...
       : mockContactInformation,
     session,
+    veteranStatus: {},
   };
 
   if (meta && veteranStatus === null) {
     const errorStatus = meta.errors.find(
       error => error.externalService === commonServices.EMIS,
     ).status;
-    userState.veteranStatus = getErrorStatusDesc(errorStatus);
```
Comment (erikphansen) [inline src/platform/user/profile/utilities/index.js:97]: 👍  setting `veteranStatus` to a string was never a good thing, IMO.

File: src/platform/user/profile/utilities/index.js [hunk 5]
```diff
@@ -88,20 +88,16 @@ export function mapRawUserDataToState(json) {
...
+    userState.veteranStatus.status = getErrorStatusDesc(errorStatus);
   } else {
-    userState.isVeteran = veteranStatus.isVeteran;
```
Comment (erikphansen) [inline src/platform/user/profile/utilities/index.js:99]: Thanks for cleaning this up! It's so weird having props from `veteranStatus` sprinkled all over the user state!

**Review by erikphansen — CHANGES_REQUESTED** [review]
Just one place where mock data in a test was changed when it didn't need to be. Otherwise looks like a good improvement!

File: src/applications/veteran-id-card/containers/VeteranIDCard.jsx (hunk 3 above)
Comment (sanlouise) [reply src/applications/veteran-id-card/containers/VeteranIDCard.jsx]: Updated! 💯

File: src/applications/veteran-id-card/containers/VeteranIDCard.jsx (hunk 2 above)
Comment (sanlouise) [reply src/applications/veteran-id-card/containers/VeteranIDCard.jsx]: Done!

File: src/applications/personalization/rated-disabilities/tests/components/RatedDisabilityView.unit.spec.jsx (hunk 1 above)
Comment (sanlouise) [reply src/applications/personalization/rated-disabilities/tests/components/RatedDisabilityView.unit.spec.jsx:11]: You're right, good catch! Updated 👩‍💻

**Review by erikphansen — APPROVED** [review]
🎉

**Review by rianfowler — APPROVED** [review]

**Pull request merged**